### PR TITLE
[Redesign] UI polish for Redeem view

### PR DESCRIPTION
**Pull request by emreboga**
This PR addresses the UI polish requests which includes a redesign of the Circular Progress bar for both determinate and indeterminate cases (see below). It also cleans up a few more React warnings/errors.

You can see the update for smoother progress bar transition here:
https://www.loom.com/share/67634ad8fceb4d99a82804b39cefb3eb?sid=ba85d219-3077-4940-befc-6a48227d0af9

Screenshots for all UI states:

Screenshot 2024-10-02 at 3 42 11 PM

Screenshot 2024-10-02 at 3 49 46 PM

Screenshot 2024-10-02 at 5 10 23 PM

Screenshot 2024-10-02 at 5 15 23 PM

### CODE REVIEW

File: wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx [hunk 1]
```diff
@@ -207,7 +207,6 @@ const AmountInput = (props: Props) => {
                 e.currentTarget.blur();
               },
               step: '0.1',
-              pattern: '[0-9]+([.|,][0-9]{1,2})?',
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx:210]: We don't need this anymore as we are checking the input type/format in the inner Input control.

File: wormhole-connect/src/views/v2/Redeem/index.tsx [hunk 2]
```diff
@@ -276,19 +292,101 @@ const Redeem = () => {
...
+    }
+
+    return (timePassed / eta) * 100;
+  }, [eta, txTimestamp, seconds]);
+
+  // In-progress circular progress bar
+  const etaCircularProgress = useMemo(() => {
+    return (
+      <>
+        <svg width={0} height={0}>
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Redeem/index.tsx:330]: This is the linear gradient we define in DOM to use in the circular progress bar below for gradient effect. That's why the svg doesn't need to be shown, it's only the def we need.

File: wormhole-connect/src/views/v2/Redeem/index.tsx [hunk 3]
```diff
@@ -276,19 +292,101 @@ const Redeem = () => {
...
+    // Check if eta has already
+    if (timePassed >= eta) {
+      return 100;
+    }
+
+    return (timePassed / eta) * 100;
+  }, [eta, txTimestamp, seconds]);
+
+  // In-progress circular progress bar
+  const etaCircularProgress = useMemo(() => {
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Redeem/index.tsx:327]: I wasn't sure but can move this progress bar in its own file to make this one smaller. It's not used anywhere else though.

File: wormhole-connect/src/views/v2/Redeem/index.tsx (hunk 2 above)
Comment (artursapek) [reply wormhole-connect/src/views/v2/Redeem/index.tsx:330]: this is pretty 1337 I gotta say well done

**Review by artursapek — APPROVED** [review]

**Pull request merged**